src/semantics/errors.rs

```diff
@@ -192,6 +192,16 @@ pub enum TypeError {
         expected_label: syntax::Label,
         found: Box<concrete::Term>,
     },
+    #[fail(
+        display = "Mismatched record size: expected {} fields but found {}",
+        expected_size,
+        found_size,
+    )]
+    RecordSizeMismatch {
+        span: ByteSpan,
+        found_size: u64,
+        expected_size: u64,
+    },
```

Comment:
We could perhaps do this to produce some better errors:

```rust
    RecordFieldMismatch {
        span: ByteSpan,
        found_labels: Vec<Label>,
        expected_labels: Vec<Label>,
    },
```

But I don't mind making a new issue, seeing as this PR fixes the immediate bug!